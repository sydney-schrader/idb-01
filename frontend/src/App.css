.App {
  text-align: center;
}


.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #e092e2;
}

.background {
  background-image: url("./LAbackground.jpg");
  background-size: contain;
  min-height: 110vh;
  display: flex;
}

.h1 {
  font-weight: bold;
  color: #fff;
  font-size: 32px;
  text-align: center;
}

