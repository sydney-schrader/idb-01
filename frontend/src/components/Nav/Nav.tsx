import React from 'react';
import styles from './Nav.module.css';
import * as data from './links.json';
import logo from "../../assets/lahhLogo.png"
const linksString = JSON.stringify(data);
const links = JSON.parse(linksString).links;

type Link = {
    label: string;
    href: string;
};

// get the links data from the links.json and iterativley adds them
const Links: React.FC<{ links: Link[] }> = ({ links }) => {
    return (
        <div className={styles['links-container']}>
            {links.map((link: Link) => {
                return (
                    <div key={link.href} className={styles['link']}>
                        <a href={link.href}>
                            {link.label}
                        </a>
                    </div>
                )
            })}
        </div>
    )
};

const Nav: React.FC<{}> = () => {
    return (
        <nav className={styles.navbar}>
            <div className={styles['logo-container']}>
            <a href="/" style={{textDecoration: 'none'}}>
            <img src= {logo} alt="Logo" style={{height: "10%", width: "10%"}} />
            <span>LACountyHomelessHelper</span></a>
            </div>
            <Links links={links} />
        </nav>
    )
}


export default Nav;