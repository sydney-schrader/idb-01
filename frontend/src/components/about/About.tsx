import React, { useState, useEffect } from "react";
//import axios from "axios";
import axios from "axios";
import { Container, Col } from 'react-bootstrap'
import jamie from '../../assets/jamie.jpg'
import pavan from '../../assets/pavan.jpg'
import sydney from '../../assets/sydney.jpg'
import zach from '../../assets/zach.jpg'
import bootstrapPic from '../../assets/bootstrapPic.jpg'
import gitlabPic from '../../assets/gitlabPic.png'
import styles from './About.module.css'
import Developer from "./Developer";
import Tool from "./Tool";
import arcadia from '../../assets/arcadia.jpg'
import volunteer from '../../assets/volunteer.jpg'
import ssa from '../../assets/ssa.jpg'
import reactPic from '../../assets/React-icon.jpg'
import flaskPic from '../../assets/flaskPic.jpg'
import sqlAlchPic from '../../assets/sqlAlchPic.jpg'
import postmanPic from '../../assets/postmanPic.jpg'

import {
  Typography,
  Stack,
} from "@mui/material";
import ApiTool from "./ApiTool";


// Developer card info

interface DevInfo {
  name: string, 
  imagePath: string, 
  devType:string,
  description:string,
  commits:number,
  issues: number,
}

const people: DevInfo[] = [
  { name: "Jamie Wong", imagePath: jamie, devType: "Frontend Developer", description: 
    "Im a senior computer science student at UT Austin. I'm from Houston, Texas and I like to read and play poker",
    commits: 0, 
    issues: 0,
  },
  { name: "Sydney Schrader", imagePath: sydney, devType: "Frontend Developer", description:
    "I'm a Junior from Austin, TX. I like to go to Barton Springs and read in my free time.",
    commits: 0, 
    issues:0, 
  },
  { name: "Zachary Voltz", imagePath: zach, devType: "Backend Developer", description:
    "I'm a senior CS student from Houston, Texas, and I've been programming since I was 9. Outside of programming, I like to read manga and rock climb",
    commits: 0,
    issues:0,
  },
  { name: "Pavan Marathi", imagePath: pavan, devType: "Backend Developer", description:
    "I'm a Junior from Houston, TX. I enjoy rock climbing and board games in my spare time.",
    commits: 0, 
    issues:0, 
  }
];





// Tool card info
interface ToolInfo {
  name: string, 
  imagePath: string, 
  description:string,
  url:string,
}

const apiTools: ToolInfo[] = [
  { name: "City Data", imagePath: arcadia, description: "Api we used to get city data", 
    url: "https://geohub.lacity.org/datasets/lacounty::homeless-counts-2020/explore?location=33.759300%2C-117.328488%2C8.29&showTable=true"
  }, 
  { name: "Resource Data", imagePath: volunteer, description: "Api we used to get resource data", 
    url: "https://geohub.lacity.org/datasets/lacounty::homeless-shelters-and-services/api"
  }, 
  { name: "Medical Data", imagePath: ssa, description: "Api we used to get medical locations", 
    url: "https://geohub.lacity.org/datasets/lacounty::medicare-and-medicaid-offices/api" 
  }
];

const devTools: ToolInfo[] = [
  { name: "Gitlab", imagePath: gitlabPic, description: "Development Platform", 
    url: "https://gitlab.com/sydneyschrader/cs373-idb-01"
  }, 
  { name: "React", imagePath: reactPic, description: "Web UI Library", 
    url: "https://react.dev/"
  }, 
  { name: "Bootstrap", imagePath: bootstrapPic, description: "CSS Framework", 
    url: "https://getbootstrap.com/"
  },
  { name: "Flask", imagePath: flaskPic, description: "Micro Web Framework", 
    url: "https://flask.palletsprojects.com/en/3.0.x/"
  },
  { name: "SQLAlchemy", imagePath: sqlAlchPic, description: "SQL Toolkit", 
    url: "https://www.sqlalchemy.org/"
  },
  { name: "Postman", imagePath: postmanPic, description: "API documentation", 
    url: "https://documenter.getpostman.com/view/28474521/2s9YJZ3jad"
  },
];


const About: React.FC<{}> = () => {

  

  const [commitData, setCommitData] = useState<any[]>([])

  const [issueData, setIssueData] = useState<any[]>([])







  useEffect(() => {
    // Get issues and commits from gitlab api
    axios.get(`https://gitlab.com/api/v4/projects/50431924/repository/commits?per_page=500`)
    .then((response) => { 
        console.log(response.data);
        setCommitData(response.data); });

    axios.get(`https://gitlab.com/api/v4/projects/50431924/issues?per_page=500`)
    .then((response) => {setIssueData(response.data); });
    
  }, []);
  

  people[0].commits = commitData.filter(
    (commit) => commit.author_name === "Jamie Wong"
  ).length

  people[0].issues = issueData.filter(
    (issue: any) => issue.closed_by != null
  ).filter(((issue) => issue.closed_by.name === "Jamie Wong")).length

  people[1].commits = commitData.filter(
    (commit) => commit.author_name === "Sydney Schrader"
  ).length

  people[1].issues = issueData.filter(
    (issue: any) => issue.closed_by != null
  ).filter(((issue) => issue.closed_by.name === "Sydney Schrader")).length

  people[2].commits = commitData.filter(
    (commit) => commit.author_name === "Zachary Voltz"
  ).length

  people[2].issues = issueData.filter(
    (issue: any) => issue.closed_by != null
  ).filter(((issue) => issue.closed_by.name === "Zachary Voltz")).length

  people[3].commits = commitData.filter(
    (commit) => commit.author_name === "Pavan Marathi"
  ).length

  people[3].issues = issueData.filter(
    (issue: any) => issue.closed_by != null
  ).filter(((issue) => issue.closed_by.name === "Pavan Marathi")).length
   
   

    return (
        <Container>
        
        <Col>
            <h1>About LA Homeless Helper</h1>
            <div className= {styles['textType']}>
            California has the largest amount of homless in the United States, the majority being in LA County. 
            This website links together city homeless data, homeless resource data, and medicare/medicaid data. 
            This disparate data can be used to analyze what locations have what services, and could be used to see what
            locations could use more help based on homless population density. This tool is designed
            for people wanting to help the homeless of LA County and also a tool 
            for homeless people to see what resources are available for them.
            </div>
 
        </Col>

        <Stack
        direction="row"
        justifyContent="center"
        >
          <div className="card-group">
            {people.map(Developer)}
          </ div>
        </Stack>


        <Col>

        <Stack
        direction="row"
        spacing={2}
        justifyContent="center"
        sx={{ padding: "20px" }}
        >
          <Typography gutterBottom variant="h4" component="div" align='center'>
            LA Homeless Helper API Sources
          </Typography>
        </Stack>

        <Stack
        direction="row"
        spacing={2}
        justifyContent="center"
        sx={{ padding: "36px" }}
        >
          {apiTools.map(ApiTool)}
        </ Stack>
  

        <Stack
        direction="row"
        spacing={2}
        justifyContent="center"
        sx={{ padding: "20px" }}
        >
          <Typography gutterBottom variant="h4" component="div" align='center'>
            Tools
          </Typography>
        </Stack>

        <Stack
        direction="row"
        spacing={2}
        justifyContent="center"
        sx={{ padding: "36px" }}
        >
          <div className="card-group">
          {devTools.map(Tool)}
          </div>
          
        </ Stack>



        </Col>

      </Container>
    );
    
   
};
 
export default About;
