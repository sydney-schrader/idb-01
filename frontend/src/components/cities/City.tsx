import React from "react";
import {  Card, Button } from 'react-bootstrap'
import arcadia from '../../assets/arcadia.jpg'
import Highlighter from "react-highlight-words";

interface CityProps {
  card: any;  // Replace 'any' with a more specific type if possible
  index: number;
  highlight: string;
}


const City: React.FC<CityProps> = ({ card, index, highlight = null }) => {
  const highlightStyle = {
    padding: 0, 
  };
  
    return(
        <Card style={{ alignItems: 'center', width: '18rem', padding: "0 10px"}} key={index} className="box">
                <Card.Title className='header-1'>
                  <b>
                      <Highlighter
                        highlightClassName="highlighter"
                        highlightStyle={highlightStyle}
                        searchWords={highlight?.split(" ") ?? []}
                        autoEscape={true}
                        textToHighlight={card.csa_label}
                      />
                  </b>
                </Card.Title>
                <img
                src={card.image_url || arcadia}
                alt={card.csa_label}
                className='card-image-top'
                style={{ width: '100%' }}
                onError={(e: React.SyntheticEvent<HTMLImageElement>) => {
                  const target = e.target as HTMLImageElement;
                  target.src = arcadia; // Set the fallback image when an error occurs
                }}
                />
                <Card.Body className="text-center">
                
                <p >
                  Unsheltered population: {card.total_unsheltered_pop} <br/>
                  Sheltered population: {card.total_sheltered_pop} <br/>
                  Total homeless population: {card.total_pop} <br/>
                  Square miles of city: {card.square_miles}<br/>
                  Density of total homeless population: {card.density_total} 
              </p>
                  <Button name={card.csa_label} href={`/cities/${card.csa_label}`} className='card-link '>
                    View {card.csa_label}
                  </Button>
                </Card.Body>
              </Card>
      )
}

export default City;