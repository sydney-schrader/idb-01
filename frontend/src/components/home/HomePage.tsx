import React from 'react';
import './home.css'
import { Container, Row, Col } from 'react-bootstrap'
import Card from 'react-bootstrap/Card'
import Button from 'react-bootstrap/Button'
import medicarePic from '../../assets/medicare-image.jpg'
import citiesPic from '../../assets/cities-image.jpg'
import resourcesPic from '../../assets/resources-image.jpg'


function Home() {
    return (
        <>
        <div className='background'>
            <h1 className='title'>
                Los Angeles <br/> Homeless Helper
            </h1>
        </div>
        <div>
           <h1 className='lower-title'>
           Instances
           </h1>
        </div>
        <Container>
        <Row>
          <Col>
            <Card style={{ alignItems: 'center' }}>
              <Card.Title className='header-1'>
                <b>Cities</b>
              </Card.Title>
              <img
                src={citiesPic}
                alt=""
                className='card-image-top'
                style={{
                  width: '100%',
                }}
              ></img>
              <Card.Body>
                <Button name='homeCities' href='/cities' className='card-link'>
                  Explore Cities in LA
                </Button>
              </Card.Body>
            </Card>
          </Col>
          <Col>
            <Card style={{ alignItems: 'center' }}>
              <Card.Title className='header-1'>
                <b>Resources</b>
              </Card.Title>
              <img
                src={resourcesPic}
                alt=""
                className='card-image-top'
                style={{
                  width: '90%',
                }}
              ></img>
              <Card.Body>
                <Button name='homeResources' href='/resources' className='card-link'>
                  Explore Resources
                </Button>
              </Card.Body>
            </Card>
          </Col>
          <Col>
            <Card style={{ alignItems: 'center' }}>
              <Card.Title className='header-1'>
                <b>Medical</b>
              </Card.Title>
              <img
                src={medicarePic}
                alt=""
                className='card-image-top'
                style={{
                  width: '100%',
                }}
              ></img>
              <Card.Body>
                <Button name='homeMedical' href='/medical' className='card-link'>
                  Explore Medical Options
                </Button>
              </Card.Body>
            </Card>
          </Col>
        </Row>
      </Container>
      </>
    );
    
   
};
 
export default Home;
