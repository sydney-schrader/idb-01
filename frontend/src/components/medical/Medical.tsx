import React from "react";
import { Card, Button } from 'react-bootstrap'
import ssa from '../../assets/ssa.jpg'
import Highlighter from "react-highlight-words";
import "./medicalCard.css"

interface MedicalProps {
  card: any;  // Replace 'any' with a more specific type if possible
  index: number;
  highlight: string;
}

const Medical: React.FC<MedicalProps> = ({ card, index, highlight = null }) => {
  const highlightStyle = {
    padding: 0, 
  };

  function handlePhones(){
    if(card.phones != null) {
      if(card.phones.length > 200) {
        card.phones = card.phones.substr(0, 120) + "\u2026";
      }
    }
    return card.phones
  }

  return(
    <Card style={{ alignItems: 'center', width: '18rem',  height: '50rem'}} key={index} className="box">
            <div className="card-title">
              <b>
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.name}
                />
              </b>
            </div>
            <img
            src={card.image_url || ssa}
            alt={card.name}
            className='card-image-top'
            style={{ width: '100%', height: '200px' }}
            onError={(e: React.SyntheticEvent<HTMLImageElement>) => {
              const target = e.target as HTMLImageElement;
              target.src = ssa; // Set the fallback image when an error occurs
            }}
            />
            <Card.Body >
            
            <p>
             { /*Name: {card.name} <br/>
              Address: {card.addrln1} <br/>
              Hours: {card.hours} <br/>
              Phone number: {card.phones}<br/>
          City: {card.city}*/}
                <div className="card-text"> Name:</div>
                
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.name}
                />
              <br/>
              
              <div className="card-text"> Address:</div> 
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.addrln1}
                />
              <br/>
              
              <div className="card-text"> Hours:</div>
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.hours || "Not Found"}
                />
              <br/>
                <div className="card-text"> Phone Number:</div>
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={handlePhones() || "Not Found"}
                />
              <br/>
                <div className="card-text"> City:</div>
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.city}
                />
              <br/>


            </p> 
              <Button name={card.name} href={`/medical/${card.name}`} className='card-link'>
                View {card.name}
              </Button>

              
            </Card.Body>
          </Card>
  )
}

export default Medical;