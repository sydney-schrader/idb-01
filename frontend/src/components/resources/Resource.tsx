import React from "react";
import {Card, Button } from 'react-bootstrap'
import volunteer from '../../assets/volunteer.jpg'
import Highlighter from "react-highlight-words";

interface ResourceProps {
  card: any;  // Replace 'any' with a more specific type if possible
  index: number;
  highlight: string;
}

const Resource: React.FC<ResourceProps> = ({ card, index, highlight = null }) => {
  const highlightStyle = {
    padding: 0, 
  };  
  
  return(
        <Card style={{ alignItems: 'center', width: '18rem', height: '50rem'}} key={index} className="box">
                <Card.Title className='header-1' style={{ width: '100%', height: '10%'}}>
                  <b>
                  <Highlighter
                        highlightClassName="highlighter"
                        highlightStyle={highlightStyle}
                        searchWords={highlight?.split(" ") ?? []}
                        autoEscape={true}
                        textToHighlight={card.name}
                      />
                  </b>
                </Card.Title>
                <img
                src={card.image_url || volunteer}
                alt={card.name}
                className='card-image-top'
                style={{ width: '100%', height: '40%'}}
                onError={(e: React.SyntheticEvent<HTMLImageElement>) => {
                  const target = e.target as HTMLImageElement;
                  target.src = volunteer; // Set the fallback image when an error occurs
                }}
                />
                <Card.Body>
                
                <p>
                  {/* Name: {card.name} <br/>
                  Address: {card.addrln1} <br/>
                  Hours: {card.hours} <br/>
                  Zip Code: {card.zip}<br/>
                  <a href={`https://${card.url}`}>{card.url ? "URL for their website" : "Website not available"}</a> */}
                  <div className="card-text"> Name:</div>
                
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.name}
                />
              <br/>
              
              <div className="card-text"> Address:</div> 
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.addrln1}
                />
              <br/>
              
              <div className="card-text"> Hours:</div>
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.hours || "Not Found"}
                />
              <br/>
                <div className="card-text"> Phone Number:</div>
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.phones || "Not Found"}
                />
              <br/>
                <div className="card-text"> City:</div>
                <Highlighter
                  highlightClassName="highlighter"
                  highlightStyle={highlightStyle}
                  searchWords={highlight?.split(" ") ?? []}
                  autoEscape={true}
                  textToHighlight={card.city}
                />
              <br/>
              </p>
                  <Button name={card.name} href={`/resources/${card.name}`} className='card-link'>
                    View {card.name}
                  </Button>
                </Card.Body>
              </Card>
      )
}

export default Resource;