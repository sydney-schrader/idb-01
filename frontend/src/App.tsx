
import './App.css';
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Nav from './components/Nav/Nav';

import About from './components/about/About';
import Cities from './components/cities/Cities';
import City1 from './components/cities/city1/City1';
import City2 from './components/cities/city2/City2';
import City3 from './components/cities/city3/City3';
import Resources from './components/resources/Resources';
import Shelter1 from './components/resources/shelter1/Shelter1';
import Shelter2 from './components/resources/shelter2/Shelter2';
import Shelter3 from './components/resources/shelter3/Shelter3';
import Medical from './components/medical/Medical';
import Office1 from './components/medical/office1/Office1';
import Office2 from './components/medical/office2/Office2';
import Office3 from './components/medical/office3/Office3';
import Home from './components/home/HomePage';


function App() {
    return (
      <>
      <BrowserRouter>
        <Nav />
        <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/home" element={<Home />} />
        <Route path="/about" element={<About />} />
        <Route path="/cities" element={<Cities />} />
        <Route path="/cities/city1" element={<City1 />} />
        <Route path="/cities/city2" element={<City2 />} />
        <Route path="/cities/city3" element={<City3 />} />
        <Route path="/resources" element={<Resources />} />
        <Route path="/resources/shelter1" element={<Shelter1 />} />
        <Route path="/resources/shelter2" element={<Shelter2 />} />
        <Route path="/resources/shelter3" element={<Shelter3 />} />
        <Route path="/medical" element={<Medical />} />
        <Route path="/medical/office1" element={<Office1 />} />
        <Route path="/medical/office2" element={<Office2 />} />
        <Route path="/medical/office3" element={<Office3 />} />
        </Routes>
       </BrowserRouter>
       </>
    );
}

export default App;

