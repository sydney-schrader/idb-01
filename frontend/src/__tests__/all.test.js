import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import City from '../components/cities/City';
import Resource from '../components/resources/Resource';
import Medical from '../components/medical/Medical';
import HomePage from '../components/home/HomePage';
import About from '../components/about/About';
import Nav from '../components/Nav/Nav';
import App from  '../App'


it('Init City', () => {
    const city = {
      csa_label: 'City of Agoura Hills',
      imageUrl: 'https://www.google.com/imgres?imgurl=https://downtownarcadia.org/wp-content/uploads/2022/10/1st-street.jpg&tbnid=tNqiVh6y5iTvGM&vet=1&imgrefurl=https://downtownarcadia.org/about-downtown-arcadia/&docid=-wjEDJJV1U95HM&w=778&h=442&source=sh/x/im/m1/1&shem=uvafe2',
      total_unsheltered_pop: 2,
      total_sheltered_pop: 0, 
      total_pop: 2,
      square_miles: 5.42772,
      density_unsheltered: 0.368479,
      density_sheltered: null, 
      density_total: 0.368479,
    };
    const component = renderer.create(
        <BrowserRouter>
          <City card={city} />
        </BrowserRouter>,
      );
    
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});

it('Init Resource', () => {
    const shelter = {
      name: '1736 Family Crisis Center',
      imageUrl: 'https://www.1736fcc.org/sites/default/files/styles/hero_icon/public/2021-06/1736%20FCC%20Logo.png?itok=gEHxVKS4',
      addrln1: '2116 Arlington Ave',
      zip: 90018, 
      hours: "Monday through Friday, 8:30am to 5:30pm. Evenings and Saturdays by appointment.",
      phones: null,
      url:"www.1736fcc.org",
      description: "The agency provides case management, counseling services, domestic violence services, family support services for low income families, runaway services, emergency and transitional shelter for battered women and their children, shelter for runaway/homeless youth, and welfare-to-work support services.  Services are provided at two community service center locations in South Los Angeles and Torrance, an emergency youth shelter in Hermosa Beach, and four confidentially located domestic violence shelters.",
    };
    const component = renderer.create(
        <BrowserRouter>
          <Resource card={shelter} />
        </BrowserRouter>,
      );
    
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});

it('Init Medicare', () => {
    const medical = {
      name: 'Alafia Mental Health Center',
      imageUrl: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRw4YFd12QcC7CwYI4hA8W7iuNWvEK47ZjU1qE9S4HKaWDdLflLgLgiz9tHq-r_UN8FzPc&usqp=CAU',
      addrln1: '11410 Avalon Blvd., ',
      zip: 90061, 
      hours: null,
      City: "City of Avalon",
      phones: '(323) 352 -6422',
      url:"www.1736fcc.org",
      description: "The agency provides case management, counseling services, domestic violence services, family support services for low income families, runaway services, emergency and transitional shelter for battered women and their children, shelter for runaway/homeless youth, and welfare-to-work support services.  Services are provided at two community service center locations in South Los Angeles and Torrance, an emergency youth shelter in Hermosa Beach, and four confidentially located domestic violence shelters.",
    };
    const component = renderer.create(
        <BrowserRouter>
          <Medical card={medical} />
        </BrowserRouter>,
      );
    
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});

it('renders Los Angeles Homeless Helper Home page', () => {
    render(<HomePage />);
    expect(screen.getByText('Los Angeles Homeless Helper')).toBeInTheDocument();
});

it('renders Instances on the Home page', () => {
    render(<HomePage />);
    expect(screen.getByText('Cities')).toBeInTheDocument();
});

it('renders about page', () => {
    render(<About />);
    expect(screen.getByText('About LA Homeless Helper')).toBeInTheDocument();
});

it('renders about page tools', () => {
    render(<About />);
    expect(screen.getByText("Tools")).toBeInTheDocument();
});

it('Init About', () => {
    const component = renderer.create(
      <BrowserRouter>
        <About />
      </BrowserRouter>,
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

  

it('Init NavBar', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


  
it('Init App', () => {
    const component = renderer.create(<App />);
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
  
  
  