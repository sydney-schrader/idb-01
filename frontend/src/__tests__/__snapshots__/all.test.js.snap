// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init City 1`] = `
<div
  className="box card"
  style={
    Object {
      "alignItems": "center",
      "padding": "0 20px",
      "width": "18rem",
    }
  }
>
  <div
    className="header-1 card-title h5"
  >
    <b />
  </div>
  <img
    className="card-image-top"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="text-center card-body"
  >
    <p>
      Unsheltered population: 
       
      <br />
      Sheltered population: 
       
      <br />
      Total homeless population: 
       
      <br />
      Square miles of city: 
      <br />
      Density of total homeless population: 
    </p>
    <a
      className="card-link  btn btn-primary"
      disabled={false}
      href="/cities/undefined"
      name="href"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      View 
    </a>
  </div>
</div>
`;

exports[`Init Medicare 1`] = `
<div
  className="box card"
  style={
    Object {
      "alignItems": "center",
      "width": "18rem",
    }
  }
>
  <div
    className="header-1 card-title h5"
  >
    <b />
  </div>
  <img
    className="card-image-top"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="card-body"
  >
    <p>
      Name: 
       
      <br />
      Address: 
       
      <br />
      Hours: 
       
      <br />
      Zip Code: 
      <br />
      <a
        href="#"
      >
        Website not available
      </a>
    </p>
    <a
      className="card-link btn btn-primary"
      disabled={false}
      href="/resources/undefined"
      name="href"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      View 
    </a>
  </div>
</div>
`;

exports[`Init Resource 1`] = `
<div
  className="box card"
  style={
    Object {
      "alignItems": "center",
      "width": "18rem",
    }
  }
>
  <div
    className="header-1 card-title h5"
  >
    <b />
  </div>
  <img
    className="card-image-top"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="card-body"
  >
    <p>
      Name: 
       
      <br />
      Address: 
       
      <br />
      Hours: 
       
      <br />
      Zip Code: 
      <br />
      <a
        href="#"
      >
        Website not available
      </a>
    </p>
    <a
      className="card-link btn btn-primary"
      disabled={false}
      href="/resources/undefined"
      name="href"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      View 
    </a>
  </div>
</div>
`;

exports[`Init Resource 2`] = `
<div
  className="box card"
  style={
    Object {
      "alignItems": "center",
      "width": "18rem",
    }
  }
>
  <div
    className="header-1 card-title h5"
  >
    <b />
  </div>
  <img
    className="card-image-top"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="card-body"
  >
    <p>
      Name: 
       
      <br />
      Address: 
       
      <br />
      Hours: 
       
      <br />
      Zip Code: 
      <br />
      <a
        href="#"
      >
        Website not available
      </a>
    </p>
    <a
      className="card-link btn btn-primary"
      disabled={false}
      href="/resources/undefined"
      name="href"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      View 
    </a>
  </div>
</div>
`;
